src/Engine/Trigger/TriggerRepeat.cpp

```diff
@@ -0,0 +1,220 @@
+        Managers().physicsManager->ForgetTriggerEnter(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->ForgetTriggerRetain(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->ForgetTriggerLeave(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->OnTriggerEnter(triggerVolume, rigidBodyComp, std::bind(&TriggerRepeat::HandleTriggerEvent, this));
+    }
+}
+
+// ADD SUPPORT FOR VECTOR
+void TriggerRepeat::OnRetain() {
+    if (!collidedEntity.empty() && collidedEntity.front()->GetComponent<Component::RigidBody>()) {
+        Component::RigidBody* rigidBodyComp = collidedEntity.front()->GetComponent<Component::RigidBody>();
+        Managers().physicsManager->ForgetTriggerEnter(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->ForgetTriggerRetain(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->ForgetTriggerLeave(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->OnTriggerRetain(triggerVolume, rigidBodyComp, std::bind(&TriggerRepeat::HandleTriggerEvent, this));
+    }
+}
+
+// ADD SUPPORT FOR VECTOR
+void TriggerRepeat::OnLeave() {
+
+    if (!collidedEntity.empty() && collidedEntity.front()->GetComponent<Component::RigidBody>()) {
+        Component::RigidBody* rigidBodyComp = collidedEntity.front()->GetComponent<Component::RigidBody>();
+        Managers().physicsManager->ForgetTriggerEnter(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->ForgetTriggerRetain(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->ForgetTriggerLeave(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->OnTriggerLeave(triggerVolume, rigidBodyComp, std::bind(&TriggerRepeat::HandleTriggerEvent, this));
+    }
+}
+
+std::string TriggerRepeat::GetName() {
+    return name;
+}
+
+void TriggerRepeat::SetName(std::string value) {
+    name = value;
+}
+
+std::vector<std::string> *TriggerRepeat::GetTargetFunction() {
+    return &targetFunction;
+}
+
+bool TriggerRepeat::GetStartActive() {
+    return startActive;
+}
+
+void TriggerRepeat::SetStartActive(bool value) {
+    startActive = value;
+}
+
+float TriggerRepeat::GetDelay() {
+    return delay;
+}
+
+void TriggerRepeat::SetDelay(float value) {
+    delay = value;
+}
+
+float TriggerRepeat::GetCooldown() {
+    return cooldown;
+}
+
+void TriggerRepeat::SetCooldown(float value) {
+    cooldown = value;
+}
+
+int TriggerRepeat::GetTriggerCharges() {
+    return triggerCharges;
+}
+
+void TriggerRepeat::SetTriggerCharges(int value) {
+    triggerCharges = value;
+}
+
+std::vector<Entity*> *TriggerRepeat::GetTargetEntity() {
+    return &targetEntity;
+}
+
+std::vector<Entity*> *TriggerRepeat::GetCollidedEntity() {
+    return &collidedEntity;
+}
+
+void TriggerRepeat::HandleTriggerEvent() {
+    triggered = true;
+}
+
+std::vector<triggerEvent::EventStruct> *TriggerRepeat::GetEventVector() {
+    return &eventVector;
+}
+
+Entity* TriggerRepeat::GetOwningEntity() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'GetOwningEntity' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)